code/game/objects/items/weapons/holy_weapons.dm

```diff
@@ -0,0 +1,328 @@
+
+/obj/item/weapon/nullrod/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is killing \himself with \the [src.name]! It looks like \he's trying to get closer to god!</span>")
+	return (BRUTELOSS|FIRELOSS)
+
+/obj/item/weapon/nullrod/attack(mob/M, mob/living/carbon/user)
+	..()
+	if(M.mind)
+		if(M.mind.vampire)
+			if(ishuman(M))
+				if(!M.mind.vampire.get_ability(/datum/vampire_passive/full))
+					to_chat(M, "<span class='warning'>The nullrod's power interferes with your own!</span>")
+					M.mind.vampire.nullified = max(5, M.mind.vampire.nullified + 2)
+
+/obj/item/weapon/nullrod/attack_self(mob/user)
+	if(reskinned)
+		return
+	if(user.mind && (user.mind.assigned_role == "Chaplain"))
+		reskin_holy_weapon(user)
+
+/obj/item/weapon/nullrod/proc/reskin_holy_weapon(mob/M)
+	var/list/holy_weapons_list = typesof(/obj/item/weapon/nullrod) - typesof(/obj/item/weapon/nullrod/fluff)
+	if(fluff_transformations.len)
+		for(var/thing in fluff_transformations)
+			holy_weapons_list += thing
+	var/list/display_names = list()
+	for(var/V in holy_weapons_list)
+		var/atom/A = V
+		display_names += initial(A.name)
+
+	var/choice = input(M,"What theme would you like for your holy weapon?","Holy Weapon Theme") as null|anything in display_names
+	if(!src || !choice || M.stat || !in_range(M, src) || M.restrained() || !M.canmove || reskinned)
+		return
+
+	var/index = display_names.Find(choice)
+	var/A = holy_weapons_list[index]
+
+	var/obj/item/weapon/nullrod/holy_weapon = new A
+
+	feedback_set_details("chaplain_weapon","[choice]")
+
+	if(holy_weapon)
+		holy_weapon.reskinned = TRUE
+		M.unEquip(src)
+		M.put_in_active_hand(holy_weapon)
+		qdel(src)
+
+/obj/item/weapon/nullrod/godhand
+	icon_state = "disintegrate"
+	item_state = "disintegrate"
+	name = "god hand"
+	desc = "This hand of yours glows with an awesome power!"
+	flags = ABSTRACT | NODROP
+	w_class = 5
+	hitsound = 'sound/weapons/sear.ogg'
+	damtype = BURN
+	attack_verb = list("punched", "cross countered", "pummeled")
+
+/obj/item/weapon/nullrod/staff
+	icon_state = "godstaff-red"
+	item_state = "godstaff-red"
+	name = "red holy staff"
+	desc = "It has a mysterious, protective aura."
+	w_class = 5
+	force = 5
+	slot_flags = SLOT_BACK
+
+/obj/item/weapon/nullrod/staff/IsShield()
+	return 1
+
+/obj/item/weapon/nullrod/staff/blue
+	name = "blue holy staff"
+	icon_state = "godstaff-blue"
+	item_state = "godstaff-blue"
+
+/obj/item/weapon/nullrod/claymore
+	icon_state = "claymore"
+	item_state = "claymore"
+	name = "holy claymore"
+	desc = "A weapon fit for a crusade!"
+	w_class = 5
+	slot_flags = SLOT_BACK|SLOT_BELT
+	sharp = 1
+	edge = 1
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
+
+/obj/item/weapon/nullrod/claymore/IsShield()
+	if(prob(30))
```

Comment:
30% chance to give you a (50 - damage/3)% chance to block 100% of incoming damage. IsShield() is weird.